src/lib/route-broadcaster.js

```diff
@@ -55,30 +55,41 @@ class RouteBroadcaster {
         throw e
       }
     }
-    setInterval(() => this.routingTables.removeExpiredRoutes(), this.routeCleanupInterval)
-    defer.setInterval(function * () {
+    this.removeExpiredRoutesSoon()
+    this.broadcastSoon()
+  }
+
+  removeExpiredRoutesSoon () {
+    setTimeout(() => {
+      this.routingTables.removeExpiredRoutes()
+      this.removeExpiredRoutesSoon()
+    }, this.routeCleanupInterval)
+  }
+
+  broadcastSoon () {
+    defer.setTimeout(function * () {
       yield this.reloadLocalRoutes()
-      this.broadcast()
-    }.bind(this), this.routeBroadcastInterval)
+      yield this.broadcast()
```

Comment:
You should probably use try/catch here and log any errors. (Same for the expired routes loop.)

For example, if `this.routingTables.toJSON` throws (line 67) even once, broadcasting would stop for good.